tutorials/tcpip/pcap-driver/main.c

```diff
@@ -189,36 +191,59 @@ static void mqtt_ev_handler(struct mg_connection *c, int ev, void *ev_data) {
              mm->data.buf, (int) mm->topic.len, mm->topic.buf));
   } else if (ev == MG_EV_CLOSE) {
     MG_INFO(("%lu CLOSED", c->id));
+    mqtt_conn = NULL;  // Mark that we're closed
   }
 }
 
 static void if_ev_handler(struct mg_tcpip_if *ifp, int ev, void *ev_data) {
-  static uint32_t ip = 0;
-  static unsigned counter = 0;
-
-  // Set initial self-assigned IP
-  if (ip == 0) ip = MG_IPV4(169, 254, 2, 100);  
-
-  // Catch ARP packets. Parse them yourself, that's easy.
-  if (ev == MG_TCPIP_EV_ARP) {
-    struct mg_str *frame = ev_data;
-    MG_INFO(("Iface %p: Got ARP frame", ifp));
-    mg_hexdump(frame->buf, frame->len);
-    // TODO: check for conflict. On conflict, increment ip and reset counter
+  // Trigger MQTT connection when we have an IP address
+  if (ifp->state == MG_TCPIP_STATE_READY &&
+      (ev == MG_TCPIP_EV_ST_CHG || ev == MG_TCPIP_EV_TIMER_1S)) {
+    struct mg_mqtt_opts opts = {.clean = true};
+    if (mqtt_conn == NULL) {
+      // mqtt_conn = mg_mqtt_connect(ifp->mgr, MQTT_URL, &opts, mqtt_ev_handler,
+      // NULL);
+      mqtt_conn = mg_mqtt_connect(ifp->mgr, MQTTS_URL, &opts, mqtt_ev_handler,
+                                  "tls enabled");
+    }
```

Comment:
We can start the MQTT connection when we get an IP, by listening the that event. Here I also listen to the _1S event in order to retry